test/fw/ptl/utils/plugins/ptl_test_runner.py

```diff
@@ -577,6 +582,87 @@ def __set_test_end_data(self, test, err=None):
+                    if k in paramkeys:
+                        if (k == 'server' or k == 'servers'):
+                            tparam_contents['servers'] = v.split(':', 1)
+                        elif (k == 'mom' or k == 'moms'):
+                            tparam_contents['moms'] = v.split(':', 1)
+                        elif k == 'comms':
+                            tparam_contents['comms'] = v.split(':', 1)
+                        elif k == 'client':
+                            tparam_contents['clients'] = v.split(':', 1)
+                        if k == 'nomom':
+                            nomomlist = v.split(':', 1)
+        for pkey in ['servers', 'moms', 'comms', 'clients']:
+            if not tparam_contents[pkey]:
+                tparam_contents[pkey] = set([shortname])
+            else:
+                tparam_contents[pkey] = set(tparam_contents[pkey])
+        if nomomlist:
+            for n in nomomlist:
+                if n in tparam_contents['moms']:
+                    tparam_contents['moms'].remove(n)
+        return tparam_contents
+
+    @staticmethod
+    def __are_requirements_matching(param_dict=None, test=None):
+        """
+        Validates test requirements against test cluster information
+        returns True on match or False otherwise
+        """
+        keylist = ['num_servers', 'num_moms', 'num_comms', 'num_clients']
+        ts_requirements = {}
+        tc_requirements = {}
+        param_count = {}
+        if test is not None:
+            test_name = getattr(test.test, '_testMethodName', None)
+            if test_name is not None:
+                method = getattr(test.test, test_name, None)
+            if method is not None:
+                tc_requirements = getattr(method, REQUIREMENTS_KEY, {})
+                cls = method.im_class
+                ts_requirements = getattr(cls, REQUIREMENTS_KEY, {})
+                if not tc_requirements:
+                    if not ts_requirements:
+                        return None
+            eff_tc_req = get_effective_reqs(ts_requirements, tc_requirements)
+        for key in ['servers', 'moms', 'comms', 'clients']:
+            param_count['num_'+key] = len(param_dict[key])
+        for pk in param_count:
+            if pk in keylist:
+                if param_count[pk] < eff_tc_req[pk]:
+                    return False
+        if set(param_dict['moms']) & set(param_dict['servers']):
+            if eff_tc_req['no_mom_on_server']:
+                return False
+        else:
+            if not eff_tc_req['no_mom_on_server']:
+                return False
+        if set(param_dict['comms']) & set(param_dict['servers']):
+            if eff_tc_req['no_comm_on_server']:
+                return False
+        else:
+            if not eff_tc_req['no_comm_on_server']:
+                return False
+        if set(param_dict['moms']) & set(param_dict['comms']):
+            if not eff_tc_req['no_comm_on_mom']:
```

Comment:
shouldn't this be "if eff_tc_req['no_comm_on_mom']" instead of "if not eff_tc_req['no_comm_on_mom']" ? Right now, it says that if there are any comms on mom hosts and no_comm_on_mom is False, then the function will return False, which seems wrong.